source/FreeRTOS_IP.c

```diff
@@ -631,6 +631,36 @@ TaskHandle_t FreeRTOS_GetIPTaskHandle( void )
  */
 void vIPNetworkUpCalls( struct xNetworkEndPoint * pxEndPoint )
 {
+    #if ( ipconfigMULTICAST_MAC_FILTERING == ipconfigENABLE )
+        IPv6_Type_t xAddressType;
+
+        if( pxEndPoint->bits.bIPv6 == pdTRUE_UNSIGNED )
+        {
+            /* Now that the network is up, pxEndPoint->ipv6_settings should hold the actual address of this
+             * end-point. For unicast addresses, generate the solicited-node multicast address that corresponds
+             * to the address and generate an MLD report for it.
+             * ToDo: Figure out what the proper place is to remove multicast addresses that are no longer valid. For
+             * example when a DHCPv6 lease expires, on network down.... etc. */
+            xAddressType = xIPv6_GetIPType( &( pxEndPoint->ipv6_settings.xIPAddress ) );
+
+            if( ( xAddressType == eIPv6_LinkLocal ) || ( xAddressType == eIPv6_SiteLocal ) || ( xAddressType == eIPv6_Global ) )
+            {
+                /* Tell the network driver to begin receiving this MAC address */
+                if( pxEndPoint->pxNetworkInterface && ( pxEndPoint->pxNetworkInterface->pfAddMulticastMAC != NULL ) )
```

Comment:
Suggestion:
~~~diff
-if( pxEndPoint->pxNetworkInterface && ( pxEndPoint->pxNetworkInterface->pfAddMulticastMAC != NULL ) )
+if( ( pxEndPoint->pxNetworkInterface != NULL ) && ( pxEndPoint->pxNetworkInterface->pfAddMulticastMAC != NULL ) )
~~~
to avoid warnings